xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -337,6 +337,108 @@ JSONRPC_STATUS CPVROperations::GetTimerDetails(const std::string &method, ITrans
   return OK;
 }
 
+CFileItemPtr CPVROperations::GetBroadcastFromBroadcastid(unsigned int broadcastid)
+{
+  EpgSearchFilter filter;
+  filter.Reset();
+  filter.m_iUniqueBroadcastId = broadcastid;
+
+  CFileItemList broadcasts;
+  int resultSize = g_EpgContainer.GetEPGSearch(broadcasts, filter);
+
+  if (resultSize != 1)
+    return CFileItemPtr();
+
+  return broadcasts.Get(0);
+}
+
+JSONRPC_STATUS CPVROperations::AddTimer(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  if (!g_PVRManager.IsStarted())
+    return FailedToExecute;
+
+  CFileItemPtr broadcast = CPVROperations::GetBroadcastFromBroadcastid(parameterObject["broadcastid"].asUnsignedInteger());
+  if (!broadcast)
+    return InvalidParams;
+
+  if (!broadcast->HasEPGInfoTag())
+    return InvalidParams;
+
+  CEpgInfoTagPtr epgTag = broadcast->GetEPGInfoTag();
+
+  if (!epgTag)
+    return InvalidParams;
+
+  bool repeating = parameterObject["repeating"].asBoolean(false);
+
+  if (epgTag->HasTimer())
+    return InvalidParams;
+
+  CPVRTimerInfoTagPtr newTimer = CPVRTimerInfoTag::CreateFromEpg(epgTag, repeating);
+  if (newTimer)
+  {
+    if (g_PVRTimers->AddTimer(newTimer))
+      return ACK;
+  }
+  return FailedToExecute;
+}
+
+
+JSONRPC_STATUS CPVROperations::DeleteTimer(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  if (!g_PVRManager.IsStarted())
+    return FailedToExecute;
+
+  CPVRTimers* timers = g_PVRTimers;
```

Comment:
`timers` is used twice. the way it's written now doesn't rely on the compiler optimising this out